homeassistant/components/freebox/router.py

```diff
@@ -0,0 +1,247 @@
+                    temperature_datas[sensor_key]["value"]
+                )
+            else:
+                # New sensor, should be unique
+                _LOGGER.debug("Adding Freebox sensor: %s", sensor_key)
+                self._temperature_sensors[sensor_key] = FreeboxSensor(
+                    self,
+                    {
+                        **TEMPERATURE_SENSOR_TEMPLATE,
+                        **{
+                            SENSOR_NAME: f"Freebox {sensor_attrs['name']}",
+                            "value": sensor_attrs["value"],
+                        },
+                    },
+                )
+                self._temperature_sensors[sensor_key].update_state(
+                    temperature_datas[sensor_key]["value"]
+                )
+
+        # Connection sensors
+        connection_datas: Dict[str, any] = await self._api.connection.get_status()
+        for sensor_key, sensor_attrs in CONNECTION_SENSORS.items():
+            if self._connection_sensors.get(sensor_key) is not None:
+                # Seen sensor -> updating
+                _LOGGER.debug("Updating Freebox sensor: %s", sensor_key)
+                self._connection_sensors[sensor_key].update_state(
+                    connection_datas[sensor_key]
+                )
+            else:
+                # New sensor, should be unique
+                _LOGGER.debug("Adding Freebox sensor: %s", sensor_key)
+                self._connection_sensors[sensor_key] = FreeboxSensor(self, sensor_attrs)
+                self._connection_sensors[sensor_key].update_state(
+                    connection_datas[sensor_key]
+                )
+
+        self._attrs = {
+            "IPv4": connection_datas.get("ipv4"),
+            "IPv6": connection_datas.get("ipv6"),
+            "connection_type": connection_datas["media"],
+            "uptime": datetime.fromtimestamp(
+                round(datetime.now().timestamp()) - syst_datas["uptime_val"]
+            ),
+            "firmware_version": self.firmware_version,
+            "serial": syst_datas["serial"],
+        }
+
+        dispatcher_send(self.hass, SENSOR_UPDATE)
+
+    async def reboot(self) -> None:
+        """Reboot the Freebox."""
+        await self._api.system.reboot()
+
+    async def close(self) -> None:
+        """Close the connection."""
+        if self._api is not None:
+            await self._api.close()
+        self._api = None
+
+    @property
+    def name(self) -> str:
+        """Return the router name."""
+        return self._name
+
+    @property
+    def manufacturer(self) -> str:
+        """Return the router manufacturer."""
+        return "Freebox SAS"
+
+    @property
+    def mac(self) -> str:
+        """Return the router MAC address."""
+        return self._mac
+
+    @property
+    def firmware_version(self) -> str:
+        """Return the router software version."""
+        return self._sw_v
+
+    @property
+    def device_info(self) -> Dict[str, any]:
```

Comment:
Can we move this to the sensor entity since it's an entity interface? Is it used in more places than in the sensors?

Reply:
In the device tracker of the router

Reply:
You mean in the switch platform? Ok, then it's ok.

Reply:
Also yep, used in : sensor + switch + device_tracker

Reply:
It's not used in the device tracker platform.